build/build.mk

```diff
@@ -87,23 +87,35 @@ define generatePlatformBuildTarget
 build-$(3)-$(1)-$(2):
   $(eval BINS_OUT_DIR_$(1)_$(2) := $(OUT_DIR)/$(1)_$(2)/$(BUILDTYPE_DIR))
 	@echo "$(ARROW) Building $(3) on $(1)/$(2) to $(BINS_OUT_DIR_$(1)_$(2))/$(3)$(BINARY_EXT)"
-	GOOS=$(1) GOARCH=$(2) go build \
+	cd $(4) && GOOS=$(1) GOARCH=$(2) go build \
```

Comment:
suggestion: rather than changing directories, pass the directory in to `go build`.

Reply:
What do you think of this idea? `go build ./my/directory` instead of `cd ./my/directory && go build`.